packages/pug-lexer/index.js

```diff
@@ -971,6 +971,31 @@ Lexer.prototype = {
     }
   },
 
+  /**
+   * EachOf.
+   */
+
+  eachOf: function() {
+    var captures;
+    if (captures = /^(?:each|for) +\[?([a-zA-Z_$][\w$]*)(?: *, *([a-zA-Z_$][\w$]*))?\]? * of *([^\n]+)/.exec(this.input)) {
```

Comment:
The way this is written allows unbalanced brackets: `each key, value] of list` as well as omiting the brackets `each key, value of list`. What I think we should have is just two possible forms, `each value of list` and `each [key, value] of list`. At the lexer/parser we can just dump the entire `[key, value]` string in the `value` property and then when we output it we'll just be treating it as JavaScript destructuring.